apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.js

```diff
@@ -1,191 +1,204 @@
+      res.redirect(res.locals.urlPrefix + '/jobSequence/' + job_sequence_id);
+    } else if (req.body.__action === 'set_time_limit') {
+      const params = {
+        assessment_instance_id: req.body.assessment_instance_id,
+        assessment_id: res.locals.assessment.id,
+        time_add: req.body.time_add,
+        time_ref: req.body.time_ref,
+        base_time: 'date_limit',
+        authn_user_id: res.locals.authz_data.authn_user.user_id,
+      };
+      if (req.body.plus_minus === 'unlimited') {
+        params.base_time = 'null';
+      } else if (req.body.plus_minus === 'expire') {
+        params.base_time = 'current_date';
+        params.time_add = 0;
+        params.time_ref = 'minutes';
+      } else if (req.body.plus_minus === 'set_total') {
+        params.base_time = 'start_date';
+      } else if (req.body.plus_minus === 'set_rem') {
+        params.base_time = 'current_date';
+      } else {
+        params.time_add *= req.body.plus_minus;
+      }
+      sqldb.queryAsync(sql.set_time_limit, params);
       res.send(JSON.stringify({}));
-    });
-  } else if (req.body.__action === 'set_time_limit_all') {
-    const params = {
-      assessment_id: res.locals.assessment.id,
-      time_add: req.body.time_add,
-      time_ref: req.body.time_ref,
-      base_time: 'date_limit',
-      reopen_closed: !!req.body.reopen_closed,
-      authn_user_id: res.locals.authz_data.authn_user.user_id,
-    };
-    if (req.body.plus_minus === 'unlimited') {
-      params.base_time = 'null';
-    } else if (req.body.plus_minus === 'expire') {
-      params.base_time = 'current_date';
-      params.time_add = 0;
-      params.time_ref = 'minutes';
-    } else if (req.body.plus_minus === 'set_total') {
-      params.base_time = 'start_date';
-    } else if (req.body.plus_minus === 'set_rem') {
-      params.base_time = 'current_date';
-    } else {
-      params.time_add *= req.body.plus_minus;
-    }
-    sqldb.query(sql.set_time_limit_all, params, function (err) {
-      if (ERR(err, next)) return;
+    } else if (req.body.__action === 'set_time_limit_all') {
+      const params = {
+        assessment_id: res.locals.assessment.id,
+        time_add: req.body.time_add,
+        time_ref: req.body.time_ref,
+        base_time: 'date_limit',
+        reopen_closed: !!req.body.reopen_closed,
+        authn_user_id: res.locals.authz_data.authn_user.user_id,
+      };
+      if (req.body.plus_minus === 'unlimited') {
+        params.base_time = 'null';
+      } else if (req.body.plus_minus === 'expire') {
+        params.base_time = 'current_date';
+        params.time_add = 0;
+        params.time_ref = 'minutes';
+      } else if (req.body.plus_minus === 'set_total') {
+        params.base_time = 'start_date';
+      } else if (req.body.plus_minus === 'set_rem') {
+        params.base_time = 'current_date';
+      } else {
+        params.time_add *= req.body.plus_minus;
+      }
+      sqldb.queryAsync(sql.set_time_limit_all, params);
```

Comment:
This is an async function; it should be awaited.

Reply:
This is now being awaited in commit [8472ae8](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/8472ae8a0869182f87c71e15b748cbd6064db927).